### PR TITLE
Add documentation for contributing a Bento iframe component

### PR DESCRIPTION
**Pull request by caroqliu**
This PR provides documentation discussing `IframeEmbed` and `ProxyIframeEmbed` as well as how/when to use them.

### CODE REVIEW

File: docs/building-a-bento-video-player.md [hunk 1]
```diff
@@ -145,7 +145,7 @@ If you need to directly insert nodes to the document, like a `<video>` element,
 
 However, it's more likely that you load a third-party iframe and you communicate with the host via `postMessage`. In this case you should use a `<VideoIframe>` as opposed to a `<VideoWrapper>`.
 
-> ⚠️ Components may **not** embed scripts from a third-party location into host documents. If a third-party script is absolutely required, like on `<amp-ima-video>`, it must be inserted in an intermediate iframe, which we call a **proxy frame**.
+> ⚠️ Components may **not** embed scripts from a third-party location into host documents. If a third-party script is absolutely required, like on `<amp-ima-video>`, it must be inserted in an intermediate iframe, which we call a [**proxy frame**](./building-a-bento-iframe-component.md).
```
Comment (alanorozco) [inline docs/building-a-bento-video-player.md:148]: Thanks for adding the cross-ref!

**Review by alanorozco — COMMENTED** [review]
I'm not sure what's the best place to put this, but we should explain the difference between a direct iframe and a proxy one. Specifically, a `VideoIframe` is not a proxy frame.

**Comment by caroqliu** [comment]
> I'm not sure what's the best place to put this, but we should explain the difference between a direct iframe and a proxy one. Specifically, a `VideoIframe` is not a proxy frame.

Added a note in [5a5dc47](https://github.com/ampproject/amphtml/pull/34911/commits/5a5dc47d9b6e8830ae78a5826502b3cf2c0e0bf4), PTAL

**Review by alanorozco — APPROVED** [review]

File: docs/building-a-bento-iframe-component.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+```js
+return <ProxyIframeEmbed frameborder="no" scrolling="no" title="My third party iframe" {...props} />
+```
+
+One important consideration is that direct iframes, such as those provided by `IframeEmbed` and [`VideoIframe`](./building-a-bento-video-player.md#loading-an-iframe-with-VideoIframe), are not the same as a **proxy iframe**, which provides an additional layer of communication between an iframe and the document. If it is not clear which is the appropriate helper for your component, [your guide](./contributing-code.md#find-a-guide) can help identify the best one to use.
+
+## Getting Started
+
+Start by generating an extension specifying `--bento` and `--nojss`. We name our extension **`amp-fantastic-embed`**, according to our [guidelines for naming a third-party component](./amp-3p-naming.md).
```
Comment (rsimha) [inline docs/building-a-bento-iframe-component.md:49]:
```suggestion
Start by generating an extension specifying `--bento` and `--nojss`. We name our extension **`amp-fantastic-embed`**, according to our [guidelines for naming a third-party component](./spec/amp-3p-naming.md).
```

Comment (caroqliu) [reply docs/building-a-bento-iframe-component.md:49]: Good catch, thank you!

File: build-system/tasks/check-links.js [hunk 3]
```diff
@@ -120,10 +120,7 @@ function checkLinksInFile(file) {
       // codepen returns a 503 for these link checks
       {pattern: /https:\/\/codepen.*/},
       // GitHub PRs and Issues can be assumed to exist
-      {
-        pattern:
-          /https:\/\/github.com\/ampproject\/amphtml\/(pull|issue)\/d+.*/,
```
Comment (rsimha) [inline build-system/tasks/check-links.js:125]: Turns out that the problem was due to the `d+`.

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/check-links.js (hunk 3 above)
Comment (caroqliu) [reply build-system/tasks/check-links.js:125]: Thanks!

**Pull request merged**